pkg/operation/botanist/component/dependencywatchdog/dependency_watchdog.go

```diff
@@ -0,0 +1,403 @@
+			},
+			Data: map[string]string{configFileName: config},
+		}
+
+		deployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      d.name(),
+				Namespace: d.namespace,
+				Labels:    d.getLabels(),
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas:             pointer.Int32(1),
+				RevisionHistoryLimit: pointer.Int32(2),
+				Selector:             &metav1.LabelSelector{MatchLabels: d.getLabels()},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{"checksum/configmap-dep-config": utils.ComputeSHA256Hex([]byte(config))},
+						Labels:      d.podLabels(),
+					},
+					Spec: corev1.PodSpec{
+						ServiceAccountName:            serviceAccount.Name,
+						TerminationGracePeriodSeconds: pointer.Int64(5),
+						Containers: []corev1.Container{{
+							Name:            name,
+							Image:           d.values.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         d.containerCommand(),
+							Ports: []corev1.ContainerPort{{
+								Name:          "metrics",
+								ContainerPort: 9643,
+								Protocol:      corev1.ProtocolTCP,
+							}},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("200m"),
+									corev1.ResourceMemory: resource.MustParse("256Mi"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("500m"),
+									corev1.ResourceMemory: resource.MustParse("512Mi"),
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{{
+								Name:      volumeName,
+								MountPath: volumeMountPath,
+								ReadOnly:  true,
+							}},
+						}},
+						Volumes: []corev1.Volume{{
+							Name: volumeName,
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: configMap.Name,
+									},
+								},
+							},
+						}},
+					},
+				},
+			},
+		}
+
+		updateMode = autoscalingv1beta2.UpdateModeAuto
+		vpa        = &autoscalingv1beta2.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      d.name() + "-vpa",
+				Namespace: d.namespace,
+			},
+			Spec: autoscalingv1beta2.VerticalPodAutoscalerSpec{
+				TargetRef: &autoscalingv1.CrossVersionObjectReference{
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+					Kind:       "Deployment",
+					Name:       deployment.Name,
+				},
+				UpdatePolicy: &autoscalingv1beta2.PodUpdatePolicy{
+					UpdateMode: &updateMode,
+				},
+				ResourcePolicy: &autoscalingv1beta2.PodResourcePolicy{
+					ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{{
+						ContainerName: autoscalingv1beta2.DefaultContainerResourcePolicy,
+						MinAllowed: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("25m"),
+							corev1.ResourceMemory: resource.MustParse("25Mi"),
```

Comment:
The probe uses a larger cache and needs a bit more memory, I think. (https://github.com/gardener/gardener/pull/4258/files#diff-aeb5d90a9661185c01f58b35703252e2f1347f1a396f98d690601c7ec9eab83dL75)

Reply:
Nice catch, fixed in 559d65b